src/clj/game/core/runs.clj

```diff
@@ -4,30 +4,224 @@
          gain-run-credits update-ice-in-server update-all-ice
          get-agenda-points gain-agenda-point optional-ability
          get-remote-names card-name can-access-loud can-steal?
-         prevent-jack-out card-flag? can-run?)
+         prevent-jack-out card-flag? can-run? run-approach-ice
+         jack-out successful-run end-run update-step corp-phase-53)
 
 ;;; Steps in the run sequence
+(defn approach-server
+  "[5]: the runner approaches the attacked server"
+  [state side eid]
+  ;; "When approached" conditionals meet their trigger condition
+  (prn "approach-server")
+  (trigger-event-sync state side (make-eid state) :approach-server (-> @state :run :server))
```

Comment:
I'm not sure how trigger-event-sync works, but should this not be waited for before proceeding?